src/app/pages/blog/article/article.js

```diff
@@ -1,130 +1,159 @@
-            image: data.article_image,
-            imageAlt: data.featured_image_alt_text,
-            title: data.heading,
-            subheading: data.subheading,
-            tags: data.tags
-        };
-    },
-    regions: {
-        sidebar: '.sticky-bit'
-    },
-    setsPageTitleAndDescription: true
+import {ShareJsx} from '~/components/share/share';
+import React, {useState, useEffect, useRef} from 'react';
+import {pageWrapper} from '~/controllers/jsx-wrapper';
+import {usePageData} from '~/helpers/controller/cms-mixin';
+import routerBus from '~/helpers/router-bus';
+
+const view = {
+    classes: ['article']
 };
 
-export default class extends componentType(spec) {
-
-    get progress() {
-        const divRect = this.el.querySelector('.body').getBoundingClientRect();
-        const viewportBottom = window.innerHeight;
-        const visibleHeight = viewportBottom - divRect.top;
-        const totalHeight = divRect.height;
-
-        if (visibleHeight <= 0) {
-            return 0;
-        }
-        if (viewportBottom >= divRect.bottom) {
-            return 100;
-        }
+function getProgress(el) {
+    const divRect = el.getBoundingClientRect();
+    const viewportBottom = window.innerHeight;
+    const visibleHeight = viewportBottom - divRect.top;
+    const totalHeight = divRect.height;
 
-        return Math.round(100 * visibleHeight / totalHeight);
+    if (visibleHeight <= 0) {
+        return 0;
     }
-
-    attachProgressRing() {
-        const progressRing = new ProgressRing({
-            radius: 45,
-            progress: this.progress,
-            stroke: 4,
-            message: this.readTime
-        });
-
-        this.regions.sidebar.attach(progressRing);
-        this.handleScroll = throttle(
-            () => {
-                progressRing.emit('update-props', {
-                    progress: this.progress
-                });
-            }
-        );
-        this.el.querySelectorAll('img')
-            .forEach((img) => {
-                img.onload = this.handleScroll;
-            });
-        window.addEventListener('scroll', this.handleScroll);
-        window.addEventListener('resize', this.handleScroll);
-        this.regions.sidebar.append(
-            new Share(
-                `${window.location.href}`,
-                'Check out this OpenStax blog article!',
-                true
-            )
-        );
+    if (viewportBottom >= divRect.bottom) {
+        return 100;
     }
 
-    attachUnits() {
-        this.update();
-        const bodyRegion = this.regionFrom('.body');
-        const bylineRegion = this.regionFrom('.byline');
-
-        this.pageData.body.forEach((unit) => {
-            bodyRegion.append(bodyUnitView(unit));
-        });
-        bylineRegion.attach(new Byline({
-            date: this.pageData.date,
-            author: this.pageData.author
-        }));
-        this.attachProgressRing();
-    }
+    return Math.round(100 * visibleHeight / totalHeight);
+}
+
```

Comment:
this could also be a hook that could use the proposed `useScrollPosition` internally.  Really it already is a hook because they're just functions, but a hook typically starts with `use`

Reply:
This one is just the helper function that figures out how far along in the article the page is. It is used by the hook.